### PR TITLE
Add custom columns to OpenAPI schema

### LINKED ISSUE #1361: Serve OpenAPI and provide useful columns in kubectl
Unfortunately, the generic columnizer was removed from kubectl due to unpredictable behavior.  We should finish the work to serve OpenAPI from the API server and allow useful columns to be shown in kubectl.

### PR DESCRIPTION
**Pull request by luksa**
Fixes #1361

This adds the required `x-kubernetes-print-columns` extensions to the OpenAPI schema, which allow kubectl to show sensible columns when listing service catalog resources:

```
$ kubectl get clusterservicebrokers
NAME         STATUS
ups-broker   FetchedCatalog

$ kubectl get clusterserviceclasses
NAME                                   EXTERNAL-NAME           BROKER       BINDABLE
4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468   user-provided-service   ups-broker   true

$ kubectl get clusterserviceplans
NAME                                   EXTERNAL-NAME   BROKER       CLASS
86064792-7ea2-467b-af93-ac9694d96d52   default         ups-broker   4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468
cc0d7529-18e8-416d-8946-6f7456acd589   premium   ups-broker   4f6e6cf6-ffdd-425f-a2c7-3c9258ad2468

$ kubectl get serviceinstances
NAME           CLASS                   PLAN      STATUS
ups-instance   user-provided-service   default   ProvisionedSuccessfully

$ kubectl get servicebindings
NAME          INSTANCE       SECRET        STATUS
ups-binding   ups-instance   ups-binding   InjectedBindResult
```

I've also added the AGE column (like the one displayed for core k8s resources), but that requires changes to kubectl, so I'll put that into a different PR.

Note:
- this PR also Includes changes from PR #1208
- to try it out, you need the to build kubectl from master, or use the `--experimental-use-openapi-print-columns` option

### CODE REVIEW

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 1]
```diff
@@ -427,6 +430,7 @@ type ExtraValue []string
 // In the future, this will be allowed and will represent the intention that
 // the ServiceInstance should have the plan and/or parameters updated at the
 // ClusterServiceBroker.
+// +k8s:openapi-gen=x-kubernetes-print-columns:custom-columns=NAME:.metadata.name,CLASS:.spec.clusterServiceClassExternalName,PLAN:.spec.clusterServicePlanExternalName,STATUS:.status.conditions[*].reason
```
Comment (pmorie) [inline pkg/apis/servicecatalog/v1beta1/types.go]: I wonder how flexible we can be here with status.  I'm not sure the current formulation of status is going to be useful - how about remove the `STATUS` bits and deal with status in a subsequent PR?

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 2]
```diff
@@ -320,6 +322,7 @@ type ClusterServicePlanList struct {
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // ClusterServicePlan represents a tier of a ServiceClass.
+// +k8s:openapi-gen=x-kubernetes-print-columns:custom-columns=NAME:.metadata.name,EXTERNAL NAME:.spec.externalName,BROKER:.spec.clusterServiceBrokerName,CLASS:.spec.clusterServiceClassRef.name
```
Comment (pmorie) [inline pkg/apis/servicecatalog/v1beta1/types.go]: I think the `CLASS` column as it is defined here is probably useful and an improvement over nothing.

With that said I think this is an area where a kubectl plugin could help immensely.

cc @jberkhahn

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 3]
```diff
@@ -225,6 +226,7 @@ type ClusterServiceClassList struct {
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // ClusterServiceClass represents an offering in the service catalog.
+// +k8s:openapi-gen=x-kubernetes-print-columns:custom-columns=NAME:.metadata.name,EXTERNAL NAME:.spec.externalName,BROKER:.spec.clusterServiceBrokerName,BINDABLE:.spec.bindable
```
Comment (pmorie) [inline pkg/apis/servicecatalog/v1beta1/types.go]: The columns proposed here are good - how able adding `plan_updateable`?

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 4]
```diff
@@ -27,6 +27,7 @@ import (
 
 // ClusterServiceBroker represents an entity that provides
 // ClusterServiceClasses for use in the service catalog.
+// +k8s:openapi-gen=x-kubernetes-print-columns:custom-columns=NAME:.metadata.name,URL:.spec.url,STATUS:.status.conditions[*].reason
```
Comment (pmorie) [inline pkg/apis/servicecatalog/v1beta1/types.go]: Not sure on the formulation of `STATUS` here, how about remove it and we'll do it in another PR?

File: pkg/apis/servicecatalog/v1beta1/types.go [hunk 5]
```diff
@@ -691,6 +695,7 @@ type ServiceBindingList struct {
 
 // ServiceBinding represents a "used by" relationship between an application and an
 // ServiceInstance.
+// +k8s:openapi-gen=x-kubernetes-print-columns:custom-columns=NAME:.metadata.name,INSTANCE:.spec.instanceRef.name,SECRET:.spec.secretName,STATUS:.status.conditions[*].reason
```
Comment (pmorie) [inline pkg/apis/servicecatalog/v1beta1/types.go]: Same concerns about `STATUS` here.

**Comment by luksa** [comment]
@pmorie Addressed all your comments & rebased to current master

**Comment by arschles** [comment]
@luksa @pmorie in your opinions, should we wait to merge this until an official kubectl version is out that supports custom columns?

iirc this wouldn't break anything right now so I don't see any problem in merging now

**Comment by luksa** [comment]
This can be merged immediately, as it has no effect until you enable the serving of the OpenAPI schema in the svc cat API server through the `--serve-openapi-spec` flag (see https://github.com/kubernetes-incubator/service-catalog/pull/1612).

**Comment by n3wscott** [comment]
LGTM
